awx/main/tasks.py

```diff
@@ -372,16 +391,29 @@ def gather_analytics():
             if acquired is False:
                 logger.debug('Not gathering analytics, another task holds lock')
                 return
-            try:
-                tgz = analytics.gather()
-                if not tgz:
-                    return
-                logger.info('gathered analytics: {}'.format(tgz))
-                analytics.ship(tgz)
-                settings.AUTOMATION_ANALYTICS_LAST_GATHER = gather_time
-            finally:
-                if os.path.exists(tgz):
-                    os.remove(tgz)
+            subset = list(all_collectors().keys())
+            incremental_collectors = []
+            for collector in expensive_collectors():
+                if collector in subset:
+                    subset.remove(collector)
+                    incremental_collectors.append(collector)
+
+            # Cap gathering at 4 weeks of data if there has been no data gathering
+            since = last_time or (gather_time - timedelta(weeks=4))
+
+            if incremental_collectors:
+                start = since
+                until = None
+                while start < gather_time:
+                    until = start + timedelta(hours = 4)
```

Comment:
@benthomasson @jctanner @wenottingham just to make sure, the probability we'd blow a /tmp disk size with 4h chunk is close to 0, right?

Reply:
Not zero, but less than we're doing now with a potential 4 week query.

Reply:
We could move this to a temporary directory under /var/lib/awx, but at a certain point if you're running out of space there's not much we can do.

Reply:
so let's try to see how this performs in the perf tests, it would have to be quite huge env so the 4h goes over /tmp (cc @kdelee )

And as another PR, we could try what @benthomasson mentioned I think, which is pagination. Using marker based pagination always getting e.g. max 100k events/jobs/etc. (the marker should be something unique like `created, id` combination, that we can use as a cursor moving through data , but maybe just `created` will work, it would get stuck if we'd have 100k events with the same created :-) ) Or we could try actual PG cursor.

In any way, the pagination will help us control max records we put under /tmp . (and the loading of the data will be probably more effective)